src/entity/specs.ts

```diff
@@ -399,4 +399,74 @@ export const ENTITIES: Record<string, EntityDescriptor<EntitySpec>> = {
       },
     },
   },
+  siSettlement: {
+    label: "Settlement",
+    collectionId: "oracle_collection:sundered_isles/settlement",
+    nameGen: (ent) =>
+      `${(ent.name_cursed && ent.name_cursed.length) ? ent.name_cursed[0].simpleResult : ent.name[0]?.simpleResult}`,
+    spec: {
+      region: {
+        id: "oracle_rollable:sundered_isles_supp/core/region",
+        firstLook: true,
+        definesAttribute: {
+          order: 1,
+          mechanism: AttributeMechanism.Snakecase,
+        },
+      },
+      location: {
+        id: "oracle_rollable:sundered_isles/settlement/location",
+        firstLook: true,
+        definesAttribute: {
+          order: 2,
+          mechanism: AttributeMechanism.Snakecase,
+        },
+      },
+      size: {
+        name: "Size",
+        id: "oracle_rollable:sundered_isles/settlement/size/{{region}}",
+        firstLook: true,
+      },
+      name: {
+        firstLook: true,
+        id: "oracle_rollable:sundered_isles/settlement/name"
+      },
+      name_cursed: {
```

Comment:
since we don't support the cursed tables yet on entities, I've been omitting the cursed tables altogether. I'd rather keep doing that until #349 is resolved.